### PR TITLE
Refactor tests for polymorphic iteration.

### PR DESCRIPTION
**Pull request by yaymukund**
@devmanhinton, @bglusman I just refactored some tests last week. I'm going to drop this now since it's bug week, but I think it's GTM. Please review.

### CODE REVIEW

File: test/polymorphic_looping.js [hunk 1]
```diff
@@ -1,94 +1,60 @@
...
-    expect($(".things- li div:nth-child(2)").html()).to.be("cool");
+var expect = require('expect.js'),
+    fs = require('fs'),
+    Backbone = require('backbone'),
+    generateTemplate = require('./support/generate_template');
+
+describe('A collection template with polymorphic attributes', function() {
+  var markup =
+    '<ul class="characters-">'+
+    '  <li class="character- rolePolymorphic-">'+
```
Comment (devmanhinton) [inline test/polymorphic_looping.js]: This is old style iteration 'characters-' 'character-' as opposed to new style 'data-each'

File: test/polymorphic_looping.js [hunk 2]
```diff
@@ -1,94 +1,60 @@
...
-  });
-  it("will interpolate the values", function(){
-    expect($('.jobPolymorphic- div:nth-child(1)').html()).to.be('Scientists welcome you, signed by <div class="firstName-">Bill</div>');
-    expect($('.jobPolymorphic- div:nth-child(2)').html()).to.be('I am your teacher and I am <div class="kind-">tough</div>');
-    expect($('.jobPolymorphic- div:nth-child(3)').html()).to.be('Default here: <div class="name-">Joe</div>');
-    expect($('.jobPolymorphic- div:nth-child(4)').html()).to.be('Default here: <div class="name-">Rando</div>');
+    it('updates the case if the switch value changes', function() {
+      expect($('.superman.whenSuperhero-')).not.to.be.empty();
+      characters.findWhere({name: 'Superman'}).set('role', 'superhero');
+      expect($('.superman.whenSupervillain-')).not.to.be.empty();
```
Comment (devmanhinton) [inline test/polymorphic_looping.js:55]: Did you mean to do

``` js
characters.findWhere({name: 'Superman'}).set('role', 'supervillain');
```

**Comment by yaymukund** [comment]
Fixed both of those things. The `set('role', 'superhero')` typo didn't break the test, which is a bit worrisome.

**Comment by devmanhinton** [comment]
@yaymukund Yeah... I was thinking that we should figure that out before merging maybe?

**Comment by yaymukund** [comment]
@devmanhinton It breaks as expected with the updated syntax. I'm inclined to put the effort toward removing the deprecated syntax entirely (and maybe opening an issue just so we have a record of the possible bug.)

**Comment by devmanhinton** [comment]
Ahhh I see. 

Agreed, if it only happens with the old syntax I agree.

**Pull request merged**